uiowa/uiowa
# Upgrade Drupal to 9.2.6 and related contributed modules

Resolves https://github.com/uiowa/uiowa/issues/4163
Resolves https://github.com/uiowa/uiowa/issues/4195
Resolves #4173 

Test the functionality addressed by the patches removed: https://github.com/uiowa/uiowa/compare/drupal-nine-two?expand=1#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34

Only core has update hooks.

<details>
<summary>Notable core diffs</summary>

- Replacements for deprecated, updates due to dependencies
- The tour module previously depended on jQuery Joyride. Joyride is now replaced with Shepherd. The core/jquery.joyride is now deprecated and core/shepherd is added. Tour config has changed, it no longer uses attributes with a data-id or data-class to indicate a target element for the tour. The selector property is used instead.
A BC layer is provided so that existing tours will still function, and any themes which use Stable or Stable9 as a base will have consistent markup for the Tour so that no theming changes should be necessary.
- CKEditor has been updated
- Sortable has been updated
- QuickEdit is being moved to contrib (death sentence?)
- Support for YouTube playlists

</details>

Contrib updates:
- search_api_autocomplete
- field_group
- layout_builder_lock
- metatag
- entity_embed
- core-composer-scaffold
- media_entity_twitter
- weight
- tvi
- views_ical
- smart_date

# ToDo

- [x] Fix sppa.uiowa.edu site split
  - _"There are some weird config issues going on with this split, but @pyrello said to ignore until we have a better idea of any changes in core and until we update config_split."_ - https://github.com/uiowa/uiowa/pull/4015
- [x] Fix pharmacy.uiowa.edu/views.view.people_list_block
- [x] Fix 500 error on https://grad.local.drupal.uiowa.edu/academics/thesis-defenses
  - I think it is choking on the number of events. out of the box the module does not seem to support https://fullcalendar.io/docs/validRange but when i limit by view filter, it starts to render again.
- [x] Run in DEV

# To Test

- Look for new errors
- Look for config diff issues
- Check for duplicate drag handles
- Media types (e.g. remote video) can not use all of the media view modes.
- You can still nest/change parent for menu links using node edit form
- Check admissions full nodes for article, person, student profile (entity print, moderation config updates)
- Check admissions that you can still properly weight scholarship content
- Check admissions that colleges taxonomy terms are still displaying like they are in PROD (tvi)
- Check grad full nodes for person, scholarship (metatag, person type status)
- Check grad thesis defense functionality as it relates to smart_date
- Check policy.clas for any issues with views_ical
- Check genetics people view menu links?
- Check immuno.grad research areas
- You can add a youtube playlist and have it display... OK.
- check authentication and caching in DEV
